File: app/src/main/java/info/nightscout/androidaps/data/Iob.java
package info.nightscout.androidaps.data;

/**
 * Created by mike on 05.06.2016.
 */
public class Iob {
    public double iobContrib = 0d;
    public double activityContrib = 0d;

    public Iob iobContrib(double iobContrib) {
        this.iobContrib = iobContrib;
        return this;
    }

    public Iob activityContrib(double activityContrib) {
        this.activityContrib = activityContrib;
        return this;
    }

    public Iob plus(Iob iob) {
        iobContrib += iob.iobContrib;
        activityContrib += iob.activityContrib;
        return this;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Iob iob = (Iob) o;

        if (Double.compare(iob.iobContrib, iobContrib) != 0) return false;
        return Double.compare(iob.activityContrib, activityContrib) == 0;
    }

    @Override
    public int hashCode() {
        int result;
        long temp;
        temp = Double.doubleToLongBits(iobContrib);
        result = (int) (temp ^ (temp >>> 32));
        temp = Double.doubleToLongBits(activityContrib);
        result = 31 * result + (int) (temp ^ (temp >>> 32));
        return result;
    }
}

File: core/interfaces/src/main/java/info/nightscout/interfaces/iob/Iob.kt
package info.nightscout.interfaces.iob

class Iob {

    var iobContrib = 0.0
    var activityContrib = 0.0

    fun iobContrib(iobContrib: Double): Iob {
        this.iobContrib = iobContrib
        return this
    }

    fun activityContrib(activityContrib: Double): Iob {
        this.activityContrib = activityContrib
        return this
    }

    operator fun plus(iob: Iob): Iob {
        iobContrib += iob.iobContrib
        activityContrib += iob.activityContrib
        return this
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val iob = other as Iob
        return if (iob.iobContrib.compareTo(iobContrib) != 0) false
        else iob.activityContrib.compareTo(activityContrib) == 0
    }

    override fun hashCode(): Int {
        var result: Int
        var temp: Long = java.lang.Double.doubleToLongBits(iobContrib)
        result = (temp xor (temp ushr 32)).toInt()
        temp = java.lang.Double.doubleToLongBits(activityContrib)
        result = 31 * result + (temp xor (temp ushr 32)).toInt()
        return result
    }
}